20181204-ArmandXiao.java

```diff
@@ -21,6 +21,6 @@ public String getHint(String s){
       else
         temp += "*";
       return temp;
-           
+          
```

Comment:
**Line 11 - 14**
Variable *a* store true or false depending on the comparison of the last character of guess and current. So the return string will contain only one letter. 
if (...) 
return
else if(...)
return
else
return
or
if (...) return
if (...) return
if (...) return
...
is fit your requirement in most cases.
Avoid using nested if structure.